examples/name-service/programs/name-service-without-macros/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+use std::net::{Ipv4Addr, Ipv6Addr};
+
+use anchor_lang::prelude::*;
+use borsh::{BorshDeserialize, BorshSerialize};
+use light_hasher::bytes::AsByteVec;
+use light_sdk::account_info::LightAccountInfo;
+use light_sdk::{
+    compressed_account::{LightAccount, LightAccounts},
+    instruction_data::LightInstructionData,
+    light_system_accounts,
+    verify::verify_compressed_accounts,
+    LightDiscriminator, LightHasher, LightTraits,
+};
+
+declare_id!("7yucc7fL3JGbyMwg4neUaenNSdySS39hbAk89Ao3t1Hz");
+
+#[program]
+pub mod name_service {
+    use light_hasher::Discriminator;
+    use light_sdk::account_info::convert_metas_to_infos;
+
+    use super::*;
+
+    pub fn create_record<'info>(
+        ctx: Context<'_, '_, '_, 'info, CreateRecord<'info>>,
+        inputs: Vec<u8>,
+        name: String,
+        rdata: RData,
+    ) -> Result<()> {
+        let inputs = LightInstructionData::deserialize(&inputs)?;
+        let mut account_infos = convert_metas_to_infos(&inputs.accounts, &crate::ID)?;
```

Comment:
I think `convert_metas_to_infos` is too highlevel because it forces all `account_infos` to have the same `program_id` and doesn't allow setting discriminators and addresses on the account level. It can probably be removed and replaced by manual `account_info` deserializations.

Reply:
Removed it. Now the conversion from `LightAccountMeta` to either `LightAccountInfo` or `LightAccount` is done manually.